force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -121,22 +194,35 @@ public with sharing class ClassesObjectsMethods {
      * Example: findTheScientist() should return "Alan Turing,Grace Hopper,Guido van Rossum,Stephen Hawking"
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
-    public static String findTheScientist() {
+    public static  String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
 
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
+        String finalScientists = '';
+        String gee = 'g';
+
+       
+           for (String name : scientistNames) {
+         
+              if (name.toLowerCase().contains(gee)){
+                    
+                 finalScientists += name + ',' ; // add each name with a comma
+                   }                     
+          
+            }       
+            
+         if (finalScientists.endsWith(',')) {
+                finalScientists = finalScientists.substring(0, finalScientists.length() - 1);
+            }
 
-        return null; // Replace null with the variable you used to store the result
-    }
+      return finalScientists;  // Replace null with the variable you used to store the result
```

Comment:
That's great! that it works, try to simplify by using the containsIgnoreCase() and the Join() methods.